src/test/unit/process-profile.test.js

```diff
@@ -229,3 +231,56 @@ describe('gecko counters processing', function() {
     expect(processedCounters[1].sampleGroups.samples.time).toEqual(offsetTime);
   });
 });
+
+describe('serializeProfile', function() {
+  it('should produce the same profile in a roundtrip', async function() {
+    const profile = processProfile(createGeckoProfile());
+    const serialized = serializeProfile(profile);
+    const roundtrip = await unserializeProfileOfArbitraryFormat(serialized);
+    // FIXME: Uncomment this line after resolving `undefined` serialization issue
+    // See: https://github.com/devtools-html/perf.html/issues/1599
+    // expect(profile).toEqual(roundtrip);
+
+    const secondSerialized = serializeProfile(roundtrip);
+    const secondRountrip = await unserializeProfileOfArbitraryFormat(
+      secondSerialized
+    );
+    expect(roundtrip).toEqual(secondRountrip);
+  });
+  describe('removing network urls', function() {
+    it('should remove all pages information', async function() {
+      const profile = processProfile(createGeckoProfile());
+      const serialized = serializeProfile(profile, false);
+      const roundtrip = await unserializeProfileOfArbitraryFormat(serialized);
+
+      if (roundtrip.pages) {
```

Comment:
Rather than wrapping this in an `if` block, it seems like we should throw an error if it doesn't exist. This will satisfy the checks for flow that the pages object exist. It also seems like we should have a check that the `http` text is in the original test data, e.g. something like this:

```js
it('should remove all pages information', async function() {
  const profile = processProfile(createGeckoProfile());

  for (const page of ensureExists(profile.pages)) {
    expect(page.url.includes('http')).toBe(true);
  }

  const serialized = serializeProfile(profile, false);
  const roundtrip = await unserializeProfileOfArbitraryFormat(serialized);

  for (const page of ensureExists(roundtrip.pages)) {
    expect(page.url.includes('http')).toBe(false);
  }
});
```